app/queue/egress.py

```diff
@@ -26,11 +26,11 @@ def __init__(self, config):
         self._kafka_producer = KafkaProducer(bootstrap_servers=config.bootstrap_servers)
         self._topic = config.host_egress_topic
 
-    def write_event(self, event):
-        logger.debug(f"Topic: {self._topic} => {event}")
+    def write_event(self, event, host_id):
```

Comment:
[_KafkaEventProducer_](https://github.com/tahmidefaz/insights-host-inventory/blob/65a96083e98f9cf05042ad1e73165f02a4773bff/app/queue/egress.py#L23) looks like a rather generic class. Let’s not bind it to the host logic. I’d rather call this argument _key_ just as _KafkaProducer_ does. This argument can be optional. It’s the caller’s responsibility to provide the key.

```suggestion
    def write_event(self, event, key=None):
```

Also the [_NullEventProducer.write_event_](https://github.com/tahmidefaz/insights-host-inventory/blob/65a96083e98f9cf05042ad1e73165f02a4773bff/app/queue/egress.py#L44) method interface should be compatible with this.

```python
    def write_event(self, event, key=None):
        logger.debug("NullEventProducer - logging event: %s, key: %s", event, key)
```